### PR TITLE
Avoid returning a BIP9Stats object with uninitialized values

### PR DESCRIPTION
**Pull request by practicalswift**
Uninitialized data potentially used in `rpc/blockchain.cpp`:

```
static UniValue BIP9SoftForkDesc(const Consensus::Params& consensusParams, Consensus::DeploymentPos id)
{
    ...
    const ThresholdState thresholdState = VersionBitsTipState(consensusParams, id);
    ...
    if (THRESHOLD_STARTED == thresholdState)
    {
        UniValue statsUV(UniValue::VOBJ);
        BIP9Stats statsStruct = VersionBitsTipStatistics(consensusParams, id);
        statsUV.push_back(Pair("period", statsStruct.period));
        statsUV.push_back(Pair("threshold", statsStruct.threshold));
        statsUV.push_back(Pair("elapsed", statsStruct.elapsed));
        statsUV.push_back(Pair("count", statsStruct.count));
        statsUV.push_back(Pair("possible", statsStruct.possible));
        rv.push_back(Pair("statistics", statsUV));
    }
    ...
    return rv;
}
```

Friendly ping @pinheadmz :-)

### CODE REVIEW

**Comment by sipa** [comment]
utACK bf2f4c36c8e37a29f81833505f76a99c5faa8a79

**Comment by pinheadmz** [comment]
utACK thanks! I think this would only apply to the genesis block, right? Which would never be `THRESHOLD_STARTED`... I don't know if that `pindex==NULL` code will ever run actually.

File: src/versionbits.cpp [hunk 1]
```diff
@@ -112,8 +112,12 @@ BIP9Stats AbstractThresholdConditionChecker::GetStateStatisticsFor(const CBlockI
     stats.period = Period(params);
     stats.threshold = Threshold(params);
 
-    if (pindex == NULL)
+    if (pindex == NULL) {
+        stats.elapsed = 0;
```
Comment (promag) [inline src/versionbits.cpp]: I see no problem in:
 - moving all default values to `BIP9Stats` constructor, or
 - moving these up even if they are overwritten below.

**Review by promag — COMMENTED** [review]
utACK bf2f4c3.

**Comment by TheBlueMatt** [comment]
Or maybe just initialize the stats object with an = {}?

**Comment by practicalswift** [comment]
@TheBlueMatt Thanks for the review! Fixed! :-)

**Comment by jonasschnelli** [comment]
utACK 3eb53b867153c957529484b5338d27e69de027c1

**Comment by MarcoFalke** [comment]
utACK 3eb53b8

**Comment by TheBlueMatt** [comment]
utACK

**Comment by fanquake** [comment]
trivialACK 3eb53b8

**Pull request merged**